### ISSUE TITLE
Attempt to invoke virtual method setShareIntent on a null object reference

### ISSUE THREAD

**Issue by maskaravivek** [commons-app/apps-android-commons#429]
Am able to see quite a few [crashes with this error](https://groups.google.com/d/msg/commons-app-android-private/oHJ8biLgNgU/yxj4cu-SBAAJ). Can we pick this up on top priority. 

```
USER_COMMENT=Я просто открыл свою фотографию.
ANDROID_VERSION=6.0.1
APP_VERSION_NAME=2.0
BRAND=Wileyfox
PHONE_MODEL=Wileyfox Swift

CUSTOM_DATA=
STACK_TRACE=java.lang.NullPointerException: Attempt to invoke virtual method 'void android.support.v7.widget.ShareActionProvider.setShareIntent(android.content.Intent)' on a null object reference
at fr.free.nrw.commons.media.MediaDetailPagerFragment.onCreateOptionsMenu(MediaDetailPagerFragment.java:228)
at android.support.v4.app.Fragment.performCreateOptionsMenu(Fragment.java:2186)
at android.support.v4.app.FragmentManagerImpl.dispatchCreateOptionsMenu(FragmentManager.java:2250)
at android.support.v4.app.FragmentController.dispatchCreateOptionsMenu(FragmentController.java:328)
at android.support.v4.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:363)
at android.support.v7.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:92)
at android.support.v7.app.AppCompatDelegateImplBase$AppCompatWindowCallbackBase.onCreatePanelMenu(AppCompatDelegateImplBase.java:331)
at android.support.v7.app.AppCompatDelegateImplV9.preparePanel(AppCompatDelegateImplV9.java:1357)
at android.support.v7.app.AppCompatDelegateImplV9.doInvalidatePanelMenu(AppCompatDelegateImplV9.java:1637)
at android.support.v7.app.AppCompatDelegateImplV9$1.run(AppCompatDelegateImplV9.java:129)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
at android.view.Choreographer.doCallbacks(Choreographer.java:670)
at android.view.Choreographer.doFrame(Choreographer.java:603)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5461)

at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by domdomegg** [comment]
This is the same as #413, which should have been fixed in #414 - if you get any reports with version 2.0.1 or later however do reopen this

**Comment by domdomegg** [comment]
Or in fact anything to do with share buttons later than 2.0.1
